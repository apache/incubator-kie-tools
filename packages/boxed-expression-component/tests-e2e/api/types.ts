import {
  DMN15__tConditional,
  DMN15__tContext,
  DMN15__tDecisionTable,
  DMN15__tFilter,
  DMN15__tFor,
  DMN15__tFunctionDefinition,
  DMN15__tInvocation,
  DMN15__tList,
  DMN15__tLiteralExpression,
  DMN15__tQuantified,
  DMN15__tRelation,
} from "@kie-tools/dmn-marshaller/dist/schemas/dmn-1_5/ts-gen/types";

export type BoxedLiteral = { __$$element: "literalExpression" } & DMN15__tLiteralExpression;
export type BoxedRelation = { __$$element: "relation" } & DMN15__tRelation;
export type BoxedContext = { __$$element: "context" } & DMN15__tContext;
export type BoxedDecisionTable = { __$$element: "decisionTable" } & DMN15__tDecisionTable;
export type BoxedList = { __$$element: "list" } & DMN15__tList;
export type BoxedInvocation = { __$$element: "invocation" } & DMN15__tInvocation;
export type BoxedFunction = { __$$element: "functionDefinition" } & DMN15__tFunctionDefinition;
export type BoxedFor = { __$$element: "for" } & DMN15__tFor;
export type BoxedEvery = { __$$element: "every" } & DMN15__tQuantified;
export type BoxedSome = { __$$element: "some" } & DMN15__tQuantified;
export type BoxedConditional = { __$$element: "conditional" } & DMN15__tConditional;
export type BoxedFilter = { __$$element: "filter" } & DMN15__tFilter;

export type Normalized<T> = WithRequiredDeep<T, "@_id">;

type WithRequiredDeep<T, K extends keyof any> = T extends undefined
  ? T
  : T extends Array<infer U>
    ? Array<WithRequiredDeep<U, K>>
    : { [P in keyof T]: WithRequiredDeep<T[P], K> } & (K extends keyof T
        ? { [P in K]-?: NonNullable<WithRequiredDeep<T[P], K>> }
        : T);
