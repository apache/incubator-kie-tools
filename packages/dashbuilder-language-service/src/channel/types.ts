/*
 * Copyright 2022 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { TextDocument } from "vscode-languageserver-textdocument";
import { CompletionItemKind, Range } from "vscode-languageserver-types";

export declare type DashbuilderLsNodeType = "object" | "array" | "property" | "string" | "number" | "boolean" | "null";
export declare type DashbuilderJsonPath = (string | number)[];

/**
 * The AST node used in the LanguageServices
 */
export type DashbuilderLsNode = {
  type: DashbuilderLsNodeType;
  value?: any;
  offset: number;
  length: number;
  colonOffset?: number;
  parent?: DashbuilderLsNode;
  children?: DashbuilderLsNode[];
};

export interface TranslateArgs {
  completion: object | string;
  completionItemKind: CompletionItemKind;
  currentNodeRange?: Range;
  cursorOffset: number;
  document: TextDocument;
  overwriteRange?: Range;
}
