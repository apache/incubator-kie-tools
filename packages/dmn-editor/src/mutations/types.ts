import { NodeType } from "../diagram/connections/graphStructure";
import { NODE_TYPES } from "../diagram/nodes/NodeTypes";

export enum NodeNature {
  DRG_ELEMENT = "DRG_ELEMENT",
  ARTIFACT = "ARTIFACT",
}

export const nodeNatures: Record<NodeType, NodeNature> = {
  [NODE_TYPES.inputData]: NodeNature.DRG_ELEMENT,
  [NODE_TYPES.decision]: NodeNature.DRG_ELEMENT,
  [NODE_TYPES.bkm]: NodeNature.DRG_ELEMENT,
  [NODE_TYPES.knowledgeSource]: NodeNature.DRG_ELEMENT,
  [NODE_TYPES.decisionService]: NodeNature.DRG_ELEMENT,
  [NODE_TYPES.textAnnotation]: NodeNature.ARTIFACT,
  [NODE_TYPES.group]: NodeNature.ARTIFACT,
};
