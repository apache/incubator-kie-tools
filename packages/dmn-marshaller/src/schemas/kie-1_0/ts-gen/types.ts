// This file was automatically generated

import { XmlParserTsRootElementBaseType } from "@kie-tools/xml-parser-ts";

export type KIE__tComponentsWidthsExtension = {
  ComponentWidths?: KIE__tComponentWidths[]; // from type KIE__tComponentsWidthsExtension @ KIE.xsd
};

export type KIE__tComponentWidths = {
  "@_dmnElementRef"?: string; // from type KIE__tComponentWidths @ KIE.xsd
  width?: number[]; // from type KIE__tComponentWidths @ KIE.xsd
};

export type KIE__tAttachment = {
  "@_url"?: string; // from type KIE__tAttachment @ KIE.xsd
  "@_name"?: string; // from type KIE__tAttachment @ KIE.xsd
};
