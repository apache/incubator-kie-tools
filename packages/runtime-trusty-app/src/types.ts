/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
export interface TrustyContextValue {
  config: {
    counterfactualEnabled: boolean;
    explanationEnabled: boolean;
    serverRoot: string;
    basePath: string;
    useHrefLinks: boolean;
  };
}

export type RemoteData<E, D> =
  | { status: RemoteDataStatus.NOT_ASKED }
  | { status: RemoteDataStatus.LOADING }
  | { status: RemoteDataStatus.FAILURE; error: E }
  | { status: RemoteDataStatus.SUCCESS; data: D };

export enum RemoteDataStatus {
  NOT_ASKED,
  LOADING,
  FAILURE,
  SUCCESS,
}

export interface Execution {
  executionId: string;
  executionDate: string;
  executedModelName: string;
  executionType: string;
  executionSucceeded: boolean;
  executorName: string;
}

export interface Executions {
  headers: Execution[];
  total: number;
}

export interface ExecutionRouteParams {
  executionId: string;
  executionType: string;
}

export interface ItemObject {
  name: string;
  value: ItemObjectValue;
}

export type ItemObjectValue = ItemObjectUnit | ItemObjectCollection | ItemObjectStructure;

export interface ItemObjectUnit {
  kind: "UNIT";
  type: string;
  value: string | number | boolean | Array<string | number | boolean> | null;
}

export interface ItemObjectCollection {
  kind: "COLLECTION";
  type: string;
  value: ItemObjectValue[] | null;
}

export type ItemObjectMap = { [key: string]: ItemObjectValue };

export interface ItemObjectStructure {
  kind: "STRUCTURE";
  type: string;
  value: ItemObjectMap;
}

export interface InputRow {
  inputLabel: string;
  inputValue?: ItemObject["value"];
  key: string;
  category: string;
}

export enum evaluationStatus {
  EVALUATING = "Evaluating",
  FAILED = "Failed",
  NOT_EVALUATED = "Not evaluated",
  SKIPPED = "Skipped",
  SUCCEEDED = "Succeeded",
}

export type evaluationStatusStrings = keyof typeof evaluationStatus;

export interface Outcome {
  outcomeId: string;
  outcomeName: string;
  outcomeResult: ItemObjectValue;
  evaluationStatus: evaluationStatusStrings;
  hasErrors: boolean;
  messages: string[];
}

export interface FeatureScores {
  featureName: string;
  featureScore: number;
}

export interface Saliency {
  outcomeId: string;
  featureImportance: FeatureScores[];
}

export enum SaliencyStatus {
  SUCCEEDED = "SUCCEEDED",
  FAILED = "FAILED",
}

export type SaliencyStatusStrings = keyof typeof SaliencyStatus;

export interface Saliencies {
  status: SaliencyStatusStrings;
  statusDetail: string;
  saliencies: Saliency[];
}

export interface ServiceIdentifier {
  groupId?: string;
  artifactId?: string;
  version?: string;
}

export interface ModelData {
  deploymentDate?: string;
  modelId?: string;
  name: string;
  namespace: string;
  modelVersion: string;
  dmnVersion: string;
  serviceIdentifier: ServiceIdentifier;
  model: string;
}

export interface CFSearchInput {
  name: string;
  value: CFSearchInputValue;
}

export type CFSearchInputValue = CFSearchInputUnit | CFSearchInputCollection | CFSearchInputStructure;

export interface CFSearchInputUnit {
  kind: "UNIT";
  type: string;
  fixed?: boolean;
  domain?: CFNumericalDomain | CFCategoricalDomain;
  originalValue: ItemObject["value"];
}

export interface CFNumericalDomain {
  type: "RANGE";
  lowerBound?: number;
  upperBound?: number;
}

export interface CFCategoricalDomain {
  type: "CATEGORICAL";
  categories: string[];
}

export interface CFSearchInputCollection {
  kind: "COLLECTION";
  type: string;
  value: Array<CFSearchInputValue> | null;
}

export type CFSearchInputValueMap = { [key: string]: CFSearchInputValue };

export interface CFSearchInputStructure {
  kind: "STRUCTURE";
  type: string;
  value: CFSearchInputValueMap | null;
}

export enum CFGoalRole {
  UNSUPPORTED,
  ORIGINAL,
  FIXED,
  FLOATING,
}

export interface CFGoal {
  id: string;
  name: string;
  role: CFGoalRole;
  value: ItemObjectValue;
  originalValue: CFGoal["value"];
}

export type CFResult = Array<unknown>;

export interface CFStatus {
  isDisabled: boolean;
  executionStatus: CFExecutionStatus;
  lastExecutionTime: null | string;
}

export enum CFExecutionStatus {
  COMPLETED,
  RUNNING,
  NOT_STARTED,
  FAILED,
  NO_RESULTS,
}

export type CFAnalysisResetType = "NEW" | "EDIT";

export interface CFAnalysisExecution {
  executionId: string;
  counterfactualId: string;
  maxRunningTimeSeconds: number;
}

export interface CFAnalysisResult {
  executionId: string;
  counterfactualId: string;
  type: "counterfactual";
  valid: boolean;
  status: "SUCCEEDED" | "FAILED";
  statusDetails: string;
  solutionId: string;
  isValid: boolean;
  stage: "INTERMEDIATE" | "FINAL";
  inputs: ItemObject[];
  outputs: ItemObject[];
  sequenceId: number;
}

export interface CFAnalysisResultsSets extends CFAnalysisExecution {
  goals: CFGoal[];
  searchDomains: CFSearchInput[];
  solutions: CFAnalysisResult[];
}

export type CFSupportMessage = {
  id: string;
  message: string;
};
