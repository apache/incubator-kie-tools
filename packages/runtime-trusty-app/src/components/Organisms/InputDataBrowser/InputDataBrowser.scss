@use "sass:math";

.input-browser {
  &__section-list {
    margin: 0 0 var(--pf-global--spacer--lg) 0;
    padding: var(--pf-global--spacer--sm) var(--pf-global--spacer--md);
    background-color: var(--pf-global--BackgroundColor--100);
    button {
      margin: 6px 12px 6px 0;
    }
    &__label {
      display: block;
      width: max-content;
      margin: 6px 10px 6px 10px;
      font-weight: 700;
      padding-top: var(--pf-global--spacer--form-element);
      padding-bottom: var(--pf-global--spacer--form-element);
    }
  }
  &__browser {
    background-color: #c6c6c6;
  }
  &__distribution {
    .input-data__wrap {
      padding-top: var(--pf-c-data-list__cell-cell--PaddingTop);
      &__desc {
        bottom: 10px;
      }
    }
    &__chart {
      // the only way I found to use the negative value of a css variable was to multiply it
      margin-top: calc(var(--pf-c-data-list__cell-cell--PaddingTop) * -1);
    }
  }
}

.input-list {
  &__headings {
    font-weight: bold;
  }
  &__has-chart {
    .pf-c-data-list__item-content {
      padding-bottom: 0;
    }
  }
  &__chart {
    --pf-c-data-list__cell-cell--PaddingTop: 0;
  }
}

.js-show-affecting-only .input-data--ignored {
  max-height: 0;
  overflow: hidden;
}
.input-data {
  &__item {
    transition: all 0.33s ease-out;
    //max-height: 4.5em;
    &:hover {
      background-color: #f5f5f5;
    }
  }
  &__affecting-switch {
    margin-bottom: var(--pf-global--spacer--lg);
  }
  &__wrap {
    position: relative;
    &__desc {
      display: block;
      text-transform: uppercase;
      font-size: 10px;
      line-height: 12px;
    }
  }
  &__icons {
    width: 30px;
    font-size: 1.25em;
    &__effect {
      color: var(--pf-global--info-color--100);
    }
    &__no-effect {
      color: lighten(#72767b, 40%);
    }
    &--hidden {
      visibility: hidden;
    }
  }
  &__score {
    font-weight: bold;
    color: var(--pf-global--info-color--100);
  }
}
.category {
  &--sticky {
    z-index: 1000;
  }
  &__heading {
    background-color: var(--pf-global--info-color--100) !important;
    color: #fff;
    font-weight: bold;
    .pf-c-data-list__item-content {
      padding-bottom: 12px;
    }
    .pf-c-data-list__cell {
      padding-top: 12px;
    }
  }
  &__sublist {
    border: none !important;
    &:first-child {
      margin-top: 10px;
    }
    &__item {
      padding: 4px;
      margin: 10px -4px 10px -4px;
      background-color: lighten(#73bcf7, 20%);
      &.pf-c-data-list {
        --pf-c-data-list--BorderTopWidth: 0;
      }
      .input-data__item:last-child {
        border-bottom: none;
      }
    }
  }
}

.data {
  &__label {
    font-size: math.div(15, 16) + rem;
    display: block;
    color: var(--pf-global--Color--200);
  }
}
