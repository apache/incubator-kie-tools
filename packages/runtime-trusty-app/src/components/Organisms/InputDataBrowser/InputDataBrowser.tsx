/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import React, { useEffect, useState } from "react";
import {
  Button,
  DataList,
  DataListCell,
  DataListItem,
  DataListItemCells,
  DataListItemRow,
  EmptyState,
  EmptyStateBody,
  EmptyStateIcon,
  EmptyStateVariant,
  Title,
} from "@patternfly/react-core/dist/js/components";
import { v4 as uuid } from "uuid";
import SkeletonDataList from "../../Molecules/SkeletonDataList/SkeletonDataList";
import FormattedValue from "../../Atoms/FormattedValue/FormattedValue";
import SkeletonFlexStripes from "../../Molecules/SkeletonFlexStripes/SkeletonFlexStripes";
import { OutlinedMehIcon } from "@patternfly/react-icons/dist/js/icons";
import { InputRow, ItemObject, ItemObjectValue, RemoteData, RemoteDataStatus } from "../../../types";
import "./InputDataBrowser.scss";
import { Split, SplitItem } from "@patternfly/react-core/dist/js/layouts";

type InputDataBrowserProps = {
  inputData: RemoteData<Error, ItemObject[]>;
};

const InputDataBrowser = ({ inputData }: InputDataBrowserProps) => {
  const [inputs, setInputs] = useState<ItemObject[] | null>(null);
  const [categories, setCategories] = useState<string[]>([]);
  const [viewSection, setViewSection] = useState<number>(0);

  const handleSectionSwitch = (index: number) => {
    setViewSection(index);
  };

  useEffect(() => {
    if (inputData.status === RemoteDataStatus.SUCCESS && inputData.data.length > 0) {
      const items: ItemObject[] = [];
      const categoryList = [];
      const rootSection: ItemObject = {
        name: "Root",
        value: { kind: "STRUCTURE", type: "root", value: {} },
      };
      for (const item of inputData.data) {
        if (item.value.kind === "UNIT") {
          // collecting inputs with values at root level (not containing components)
          rootSection.value.value[item.name] = item.value;
        } else {
          items.push(item);
          categoryList.push(item.name);
        }
      }
      if (Object.entries(rootSection.value.value).length !== 0) {
        // if the root section is not empty, than add the root section as first one
        items.unshift(rootSection);
        categoryList.unshift("Root");
      }
      setInputs(items);
      setCategories(categoryList);
      // open the fist section as default
      setViewSection(0);
    }
  }, [inputData]);

  return (
    <div className="input-browser">
      {inputData.status === RemoteDataStatus.LOADING && (
        <>
          <div className="input-browser__section-list">
            <SkeletonFlexStripes stripesNumber={6} stripesWidth={"100px"} stripesHeight={"1.5em"} />
          </div>
          <SkeletonDataList rowsCount={4} colsCount={6} hasHeader={true} />
        </>
      )}
      {inputData.status === RemoteDataStatus.SUCCESS && inputData.data.length > 0 && (
        <>
          <div className="input-browser__section-list">
            <Split>
              <SplitItem>
                <span className="input-browser__section-list__label">Browse sections</span>
              </SplitItem>
              <SplitItem>
                {categories.map((item, index) => (
                  <Button
                    type={"button"}
                    variant={index === viewSection ? "primary" : "control"}
                    isActive={index === viewSection}
                    key={`section-${index}`}
                    onClick={() => handleSectionSwitch(index)}
                  >
                    {item}
                  </Button>
                ))}
              </SplitItem>
            </Split>
          </div>
          <DataList aria-label="Input Data" className="input-browser__data-list">
            <DataListItem aria-labelledby="header" key="header" className="input-browser__header">
              <DataListItemRow>
                <DataListItemCells
                  dataListCells={[
                    <DataListCell width={3} key="Input Data">
                      <span>Input data</span>
                    </DataListCell>,
                    <DataListCell width={3} key="Value">
                      <span>Value</span>
                    </DataListCell>,
                    <DataListCell width={3} key="Distribution">
                      <></>
                    </DataListCell>,
                  ]}
                />
              </DataListItemRow>
            </DataListItem>
            {inputs && renderItem(inputs[viewSection].name, inputs[viewSection].value)}
          </DataList>
        </>
      )}
      {inputData.status === RemoteDataStatus.SUCCESS && inputData.data.length === 0 && (
        <EmptyState variant={EmptyStateVariant.full}>
          <EmptyStateIcon icon={OutlinedMehIcon} />
          <Title headingLevel="h5" size="lg">
            No input data
          </Title>
          <EmptyStateBody>No inputs available to display.</EmptyStateBody>
        </EmptyState>
      )}
    </div>
  );
};

const ItemsSubList = (props: { name: string; value: ItemObjectValue; listCategory: string }) => {
  const { name, value, listCategory } = props;

  return (
    <DataListItem aria-labelledby="" className={"category__sublist"}>
      <DataList aria-label="" className={"category__sublist__item"}>
        {value.kind === "UNIT" && (
          <InputValue inputLabel={name} inputValue={value} key={name} category={listCategory} />
        )}
        {value.kind === "STRUCTURE" &&
          Object.entries(value.value).map(([key, value]) => {
            return <InputValue key={key} inputLabel={key} inputValue={value} category={listCategory} />;
          })}
      </DataList>
    </DataListItem>
  );
};

const CategoryLine = (props: { categoryLabel: string }) => {
  const { categoryLabel } = props;
  const categoryKey = categoryLabel.replace(" ", "").toLocaleLowerCase();
  return (
    <DataListItem aria-labelledby={categoryLabel} key={"category-" + categoryKey} className="category__heading">
      <DataListItemRow>
        <DataListItemCells
          dataListCells={[
            <DataListCell key={categoryLabel}>
              <span>{categoryLabel}</span>
            </DataListCell>,
          ]}
        />
      </DataListItemRow>
    </DataListItem>
  );
};

const InputValue = (props: InputRow) => {
  const { inputValue, inputLabel, category } = props;
  const effectItemClass = "input-data--ignored";
  const dataListCells = [];
  dataListCells.push(
    <DataListCell width={3} key="primary content" className="input-data__wrap">
      <span>{inputLabel}</span>
      <span className="input-data__wrap__desc">{category}</span>
    </DataListCell>
  );
  dataListCells.push(
    <DataListCell width={3} key="secondary content">
      <span>
        <FormattedValue value={inputValue.value} />
      </span>
    </DataListCell>
  );
  dataListCells.push(
    <DataListCell width={3} key="distribution">
      <span />
    </DataListCell>
  );

  return (
    <DataListItem
      aria-labelledby={"Input columns"}
      key={`input-item-heading`}
      className={`input-data__item ${effectItemClass}`}
    >
      <DataListItemRow>
        <DataListItemCells dataListCells={dataListCells} />
      </DataListItemRow>
    </DataListItem>
  );
};

let itemCategory = "";

const renderItem = (name: string, value: ItemObjectValue, categoryName?: string): JSX.Element => {
  const renderedItems: JSX.Element[] = [];

  const elaborateRender = (name: string, value: ItemObjectValue, category?: string): JSX.Element => {
    if (value.kind === "UNIT") {
      return <InputValue inputLabel={name} inputValue={value} category={itemCategory} key={name} />;
    }

    if (value.kind === "STRUCTURE" || value.kind === "COLLECTION") {
      itemCategory = category ? `${itemCategory} / ${category}` : name;
      const categoryLabel = itemCategory.length > 0 ? `${itemCategory}` : name;

      if (value.kind === "STRUCTURE") {
        Object.entries(value.value).forEach(([key, value]) => {
          renderedItems.push(renderItem(key, value, key));
        });
      } else if (value.kind === "COLLECTION") {
        value.value.forEach((value) => {
          renderedItems.push(<ItemsSubList name={name} value={value} key={uuid()} listCategory={categoryLabel} />);
        });
      }

      return (
        <React.Fragment key={categoryLabel}>
          <div className="category">
            <CategoryLine categoryLabel={categoryLabel} key={`category-${categoryLabel}`} />
          </div>
          {renderedItems}
        </React.Fragment>
      );
    }
    return <></>;
  };

  return elaborateRender(name, value, categoryName);
};

export default InputDataBrowser;
