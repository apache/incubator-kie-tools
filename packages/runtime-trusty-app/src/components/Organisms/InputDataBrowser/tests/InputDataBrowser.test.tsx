/**
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
import React from "react";
import InputDataBrowser from "../InputDataBrowser";
import { mount, shallow } from "enzyme";
import { ItemObject, ItemObjectUnit, RemoteData, RemoteDataStatus } from "../../../../types";

describe("InputDataBrowser", () => {
  test("renders a loading animation while fetching data", () => {
    const inputData = { status: RemoteDataStatus.LOADING } as RemoteData<Error, ItemObject[]>;
    const wrapper = shallow(<InputDataBrowser inputData={inputData} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("renders a list of inputs", () => {
    const inputData = {
      status: RemoteDataStatus.SUCCESS,
      data: [
        {
          name: "Asset Score",
          value: {
            kind: "UNIT",
            type: "number",
            value: 738,
          },
        },
        {
          name: "Asset Amount",
          value: {
            kind: "UNIT",
            type: "number",
            value: 700,
          },
        },
        {
          name: "Property",
          value: {
            kind: "STRUCTURE",
            type: "tProperty",
            value: {
              "Purchase Price": {
                kind: "UNIT",
                type: "number",
                value: 34000,
              },
            },
          },
        },
      ],
    } as RemoteData<Error, ItemObject[]>;
    const wrapper = mount(<InputDataBrowser inputData={inputData} />);
    const dataList = wrapper.find("DataList.input-browser__data-list");

    expect(wrapper.find(".input-browser__section-list")).toHaveLength(1);
    expect(wrapper.find(".input-browser__section-list button")).toHaveLength(2);
    expect(wrapper.find(".input-browser__section-list button").at(0).text()).toMatch("Root");
    expect(wrapper.find(".input-browser__section-list button").at(1).text()).toMatch("Property");
    expect(dataList).toHaveLength(1);
    expect(dataList.find("DataListItem.input-browser__header")).toHaveLength(1);
    expect(dataList.find("CategoryLine")).toHaveLength(1);
    expect(dataList.find("CategoryLine").prop("categoryLabel")).toMatch("Root");
    expect(dataList.find("InputValue")).toHaveLength(2);
    expect((dataList.find("InputValue").at(0).prop("inputValue") as ItemObjectUnit).value).toBe(738);
    expect((dataList.find("InputValue").at(1).prop("inputValue") as ItemObjectUnit).value).toBe(700);

    wrapper.find(".input-browser__section-list button").at(1).simulate("click");
    const propertyDataList = wrapper.find("DataList.input-browser__data-list");

    expect(propertyDataList.find("DataListItem.input-browser__header")).toHaveLength(1);
    expect(propertyDataList.find("CategoryLine")).toHaveLength(1);
    expect(propertyDataList.find("CategoryLine").prop("categoryLabel")).toMatch("Property");

    expect(propertyDataList.find("InputValue")).toHaveLength(1);
    expect((propertyDataList.find("InputValue").prop("inputValue") as ItemObjectUnit).value).toBe(34000);
  });
});
