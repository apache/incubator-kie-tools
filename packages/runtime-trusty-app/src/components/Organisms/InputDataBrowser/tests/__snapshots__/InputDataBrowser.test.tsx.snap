// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputDataBrowser renders a loading animation while fetching data 1`] = `
<div
  className="input-browser"
>
  <div
    className="input-browser__section-list"
  >
    <SkeletonFlexStripes
      stripesHeight="1.5em"
      stripesNumber={6}
      stripesWidth="100px"
    />
  </div>
  <SkeletonDataList
    colsCount={6}
    hasHeader={true}
    rowsCount={4}
  />
</div>
`;
